module/User/src/Service/AclService.php

```diff
@@ -52,34 +52,19 @@ protected function createAcl()
         $this->acl->addRole(new Role('tueguest'), 'guest');
         $this->acl->addRole(new Role('user'), 'tueguest');
         $this->acl->addrole(new Role('apiuser'), 'guest');
-        $this->acl->addrole(new Role('sosuser'), 'apiuser');
         $this->acl->addrole(new Role('active_member'), 'user');
         $this->acl->addrole(new Role('company_admin'), 'active_member');
         $this->acl->addRole(new Role('admin'));
         $this->acl->addRole(new Role('photo_guest'), 'guest');
 
-        $user = $this->getIdentity();
-
-        // add user to registry
-        if ($user instanceof User) {
-            $roles = $user->getRoleNames();
-            // if the user has no roles, add the 'user' role by default
-            if (empty($roles)) {
-                $roles = ['user'];
-            }
-
-            if (count($user->getMember()->getCurrentOrganInstallations()) > 0) {
-                $roles[] = 'active_member';
-            }
-
-            $this->acl->addRole($user, $roles);
-        }
-
         // admins are allowed to do everything
         $this->acl->allow('admin');
 
-        // board members also are admins
-        $this->acl->allow('user', null, null, new IsBoardMember());
+        foreach ($this->acl->getRoles() as $role) {
+            // board members also are admins
+            // since they can have any other role, we need to apply this assertion to all roles
+            $this->acl->allow($role, null, null, new IsBoardMember());
+        }
```

Comment:
This actually breaks the `admin` role (nothing is allowed). Furthermore, it still does not give board members the appropriate privileges.